### DISCUSSION TITLE
Help, my canvas is running away from me.

### DISCUSSION THREAD

**Discussion by 9SMTM6** [emilk/egui#4660] [Q&A] [answered]
I've got no idea whats going on here:

https://github.com/emilk/egui/assets/44668330/af033b32-637e-413d-978d-5999ab9518b2

the same thing also happens in Firefox, when other issues are resolved, its just harder to see.

As you can see this is not happening on native.

This render is also a straight copy from the demo app (where the call is a bit more abstracted, but I spot no UI-elements between that could change the behavior). In the demo app it works fine on browsers too, and I just can't find what makes the difference, unless its the general setup thats broken.

The source code is [here](https://github.com/9SMTM6/mcmc-demo/tree/resize_issue). Its essentially the eframe template with wgpu as renderer (which had/has a bunch of issues, documented in [this issue](https://github.com/emilk/egui/issues/4657)) and then I removed the placeholder content and pasted in the webgpu render from the demo app.

On native the size set [here - unchanged from the demo app - ](https://github.com/9SMTM6/mcmc-demo/blob/e1fae60d0af66201d2d4951b00e8b1e8d0d62a5f/src/custom3d_wgpu.rs#L182) is respected, on web seemingly not.

**Comment by 9SMTM6** [comment, marked as answer]
I've solved this issue by transplanting CSS parts from https://github.com/emilk/egui/blob/2c00cb39912e1043e883bacfeaec5806b51b7841/web_demo/index.html#L40 until https://github.com/emilk/egui/blob/2c00cb39912e1043e883bacfeaec5806b51b7841/web_demo/index.html#L69 into the respective spot that's provided by the template.

I'll make a PR or something for the template app.

**Comment by emilk** [comment]
The next release of egui will fix this: https://github.com/emilk/egui/pull/4536
